snapcraft/storeapi/__init__.py

```diff
@@ -287,17 +287,36 @@ def _download_snap(self, name, channel, arch, download_path,
             return
         logger.info('Downloading {}'.format(name, download_path))
 
+        resume_possible = False
+        total_read = 0
+        probe_url = requests.head(download_url)
+        if probe_url.is_redirect:
+            download_url = probe_url.headers['Location']
+            resume_possible = True
+
         # HttpAdapter cannot help here as this is a stream.
         # LP: #1617765
         not_downloaded = True
         retry_count = 5
         while not_downloaded and retry_count:
-            request = self.cpi.get(download_url, stream=True)
+            headers = {}
+            if resume_possible and os.path.exists(download_path):
+                total_read = os.path.getsize(download_path)
+                headers['Range'] = 'bytes={}-'.format(total_read)
```

Comment:
Right, the specification does not require it, space should be fine, perhaps it's a specialty of internap :-/ 

Anyway, as I've noted in the previous comment, this is how it's currently working, but it's worth noting that it contains internap hacks. Other than that 👍